options/posix/generic/net-if-stubs.cpp

```diff
@@ -2,6 +2,10 @@
 #include <dirent.h>
 #include <bits/ensure.h>
 
+#ifdef __cplusplus
```

Comment:
Don't do this. The reason why they are getting exported with C++ linkage is because the file is missing a ``#include <net/if.h>``